src/stack.rs

```diff
@@ -3,12 +3,56 @@
 //
 // The implementation of SinglyLinkedList
 
+use std::mem;
+
 use Stack;
 
+struct Node<T> {
+    value: T,
+    next: Option<Box<Node<T>>>,
+}
+
+impl<T> Node<T> {
+    fn new(val: T, next: Option<Box<Node<T>>>) -> Self {
+        Node { value: val, next: next, }
+    }
+}
+
 pub struct SinglyLinkedList<T> {
-    data: T // Probably not what you want
+    size: usize,
+    front: Option<Box<Node<T>>>,
 }
 
 impl<T: Eq> Stack<T> for SinglyLinkedList<T> {
-    // ????
+    fn new() -> Self {
+        SinglyLinkedList { size: 0, front: None }
+    }
+
+    fn push_front(&mut self, item: T) {
+        let new_front = Some(Box::new(Node::new(item, mem::replace(&mut self.front, None))));
```

Comment:
If you want to `replace` an option with `None`, use `Option::take`. Super convenient :smile: !